src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,186 @@
+package duke.tasklist;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.lang.IndexOutOfBoundsException;
+
+import duke.task.Tasks;
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Todos;
+import duke.storage.Storage;
+
+public class TaskList {
+
+    private ArrayList<Tasks> taskList;
+
+    public TaskList(ArrayList<Tasks> taskList) {
+        this.taskList = new ArrayList<Tasks>();
+        this.taskList.addAll(taskList);
+    }
+
+    public TaskList() {
+        this.taskList = new ArrayList<Tasks>();
+    }
+
+    // Maybe can abstract out the printing portion
+    public String queryTasks(String query) {
+        ArrayList<Tasks> unfilteredTasks = new ArrayList<Tasks>(taskList);
+        ArrayList<Tasks> filteredTasks = new ArrayList<Tasks>();
+        StringBuilder sb1 = new StringBuilder("    Here are the matching tasks in your list: \n");
```

Comment:
Perhaps its better to not use number (`sb1`, `sb2`, etc) as discussed in the lecture video.